lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -323,6 +325,17 @@ def add_metadata(post, options = {})
         post[:metadata][:event_type] = options[:event_type] if options[:event_type]
       end
 
+      def add_card_brand(post, options)
+        return unless options[:card_brand]
+
+        card_network = {}
+        card_network[:network] = options[:card_brand]
+
+        post[:payment_method_options] ||= {}
+        post[:payment_method_options][:card] ||= {}
+        post[:payment_method_options][:card] = card_network
```

Comment:
🔍 Shouldn't this be `post[:payment_method_options][:card][:network] = options[:card_brand]` ?

I think the idea here is correct, and it mirrors the implementation elsewhere in this gateway model which tries not to clear out any existing fields as we build out the hash for various scenarios. But as this is currently written, the memoization you're doing in line 335 becomes unnecessary, because whatever is currently nested at `post[:payment_method_options][:card]` (assuming there is something there other than a newly instantiated `{}`) will be overwritten by `card_network`

Reply:
heya @jcreiff thanks for looking at this! As for your question if this should be "post[:payment_method_options][:card][:network] = options[:card_brand]" I originally tried to write it that way but ran into issues with create_setup_intent because that path does not currently have any ":payment_method_options" by default in the post. So it throws a NoMethodError. 
Looking back I do agree that line 335 is not needed as your explanation is correct. Thanks for catching that!